features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/admin-dashboard/site/blocks/theme-upload-task/block.jag

```diff
@@ -0,0 +1,76 @@
+<%
+include("/jagg/jagg.jag");
+var site = require("/site/conf/site.json");
+jagg.block("theme-upload-task", {
+    initializer: function (data) {
+
+    },
+
+    getInputs: function () {
+        return {
+
+        }
+    },
+
+    getOutputs: function (inputs) {
+        var user = jagg.getUser();
+        var log = new Log("Theme Upload Task :");
+        var error = false;
+        var success = false;
+
+        if (request.getMethod() == "POST") {
+            var themeType = request.getParameter("type");
+            var themeTypes = getCustomStoreThemeTypes();
+            if (!themeType || themeTypes.indexOf(themeType) == -1) {
+                error = "Please select the correct theme type";
+                log.error("Theme type : " + themeType + " is not found in :" + stringify(themeTypes));
+            } else {
+                if (user.userDomain != null && user.userDomain != '') {
+                    var file = request.getFile("theme");
+                    if (file) {
+                        var file_name = file.getName();
+                        var file_ext = file_name.substr(file_name.length - 4);
+                        if (file_ext != ".zip") {
+                            error = "Please upload a valid zip file.";
+                            log.error("File extension : " + file_ext + " of file : " + file_name + " is not supported");
+                        }
+                        else {
+                            var whiteListedExt = getWhiteListedFileExt();
+                            var appManager = require("thememanager");
+                            var managerHostObj = new appManager.ThemeManager();
+                            try {
+                                success =
+                                managerHostObj.addCustomTheme(file, user.userDomain, themeType, whiteListedExt);
+                            } catch (e) {
+                                error = "Error while deploying theme.";
+                                log.error(error, e);
+                            }
+                        }
+                    } else {
+                        error = "Please upload theme file.";
+                    }
+                }
+                else {
+                    error = "Tenant is not specified, unable to deploy the theme.";
+                }
+            }
+
+        }
+
+        return {
+            "error": error,
```

Comment:
Actually error should be corrected as "errorMessage",.
if operation is failed (i.e success=false) there will be error message and theme will be null.
if operation success (i.e success =true) then there will be themes obejct and errorMessage will be null

Based on the "success" state client will decide whether to process errorMessage or theme object 
{sucesss ="true/false","errorMessage: "",themes=""}

Any suggestions ?